src/index.js

```diff
@@ -492,6 +489,33 @@ class Libp2p extends EventEmitter {
     peer.protocols && this.peerStore.protoBook.set(peer.id, peer.protocols)
   }
 
+  /**
+   * Will dial a substet of the peers already known
+   * @private
+   * @returns {void}
+   */
+  _maybeConnectPeerStorePeers () {
+    if (!this._config.peerDiscovery.autoDial) {
+      return
+    }
+
+    const minPeers = this._options.connectionManager.minPeers || 0
+
+    // TODO: this should be ordered by score on PeerStoreV2
+    this.peerStore.peers.forEach(async (peer) => {
```

Comment:
@jacobheun I started by doing the dials in serial as the token based dial will already use parallel dials for several multiaddrs the peer we are targeting might have. Also, the discovery services kicked in, and their new discoveries might be more important than the older ones (this is where having scores for the older peers would help a lot too).

Another solution I thought is to do the `minPeers` parallel dials and then do the rest in serial (to compensate possible offline peers). Let me know what do you think about the pros and cons of both approaches.